src/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -61,104 +61,160 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		String siteIdentifier = getSiteIdentifier(msg);
+
+		if (alreadySeen(msg)) {

```

Comment:
The `siteIdentifier` is unique enough that we don't need to check the past history anymore (IMO).